sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/AvoidNotShortCircuitOperatorsForBooleanCheck.java

```diff
@@ -140,8 +141,11 @@ public final void visitToken(final DetailAST detailAST) {
      * @return "true" if current method or variable has a Boolean type.
      */
     private static boolean isBooleanType(final DetailAST node) {
-        return BOOLEAN.equals(CheckUtil.createFullType(
-                node.findFirstToken(TokenTypes.TYPE)).getText());
+        final FullIdent methodReturnType =
+                CheckUtil.createFullType(node.findFirstToken(TokenTypes.TYPE));
+        return methodReturnType
+                .getText()
+                .contains(BOOLEAN);
```

Comment:
Original code was `equals` and now this changed to `contains`. Doesn't this mean we will get false positives with things like `AtomicBoolean`?

Reply:
Not `AtomicBoolean`, no. It would only match strings that contain lowercase `boolean`.  You can't use logical operators on `AtomicBoolean` , anyway. This would match a return type of a class that had lowercase `boolean` in it though; but the code wouldn't compile if we used a bitwise operator on anything besides a boolean or an int anyway. So, we could have a false positive on an int that was named `myboolean`. Previously, `createFullType()` was not returning the `[]` reliably for some 1-D arrays ( same bug from the main PR), so it was a happy accident that this worked in the first place.

Reply:
I can just use a matcher here, to prevent false positives.

Reply:
Ok, `contains` is case sensitive. But then makes it a concern if a user creates a class `Funwithboolean`.

Reply:
But, you would not be able to return the result of a bitwise operator on, or with any class except `int` and `boolean`. It would not compile.

Reply:
Yes, but it is not clear from this snippet of code as that is in other parts of the class. This method just says `isBoolean` and we don't know who might make use of it in the future.

Then I just have 1 final question since you changed this to an expression. Why not add `^$` to enforce you are only looking at exactly a `boolean` (and any `[]`s)?

Reply:
done